proof/refine/ARM/Ipc_R.thy

```diff
@@ -1864,20 +1882,30 @@ lemma cancelAllSignals_weak_sch_act_wf[wp]:
 
 lemma setSchedContext_weak_sch_act_wf:
   "setSchedContext p sc \<lbrace> \<lambda>s. weak_sch_act_wf (ksSchedulerAction s) s \<rbrace>"
-  sorry
+  apply (wp weak_sch_act_wf_lift)
+  done
 
 lemma setReply_weak_sch_act_wf:
   "setReply p r \<lbrace> \<lambda>s. weak_sch_act_wf (ksSchedulerAction s) s \<rbrace>"
-  sorry
+  apply (wp weak_sch_act_wf_lift)
+  done
 
 crunches replyUnlink
   for nosch[wp]: "\<lambda>s. P (ksSchedulerAction s)"
   (simp: crunch_simps wp: crunch_wps)
 
-lemma replyUnlink_weak_sch_act_wf[wp]:
-  "replyUnlink r t \<lbrace> \<lambda>s. weak_sch_act_wf (ksSchedulerAction s) s \<rbrace>"
+(* see replyUnlink_weak_sch_act_wf in IpcCancel_R *)
+lemma replyUnlink_weak_sch_act_wf:
```

Comment:
Are you sure this one is provable? `replyUnlink` will set the thread state of `p` to `Inactive`

Reply:
I don’t think so. I actually tried removing this sorry but it requires a bit more work than I could afford for the purpose of this PR to sort out the dependency. So, yes, this needs to be sorted. I will update the comment to be more specific.

Reply:
I've added a similar lemma in my recent PR merge, but it had an additional `sch_act_not` in the precondition of the hoare triple.